agent/agent.go

```diff
@@ -796,16 +796,13 @@ func (c *configInternal) APIInfo() (*api.Info, bool) {
 		// loopback, and when/if this changes localhost should resolve
 		// to IPv6 loopback in any case (lp:1644009). Review.
 		localAPIAddr := net.JoinHostPort("localhost", strconv.Itoa(port))
-		addrInAddrs := false
+		newAddrs := []string{localAPIAddr}
```

Comment:
There is an 'if' just above:
if isController {
 ...
newAddrs :=

Which should mean that we only inject local address if this machine is already known that it wants to be a controller.